### PR TITLE
Restore partial subnet option parsing behavior

### PR DESCRIPTION
**Pull request by mattrobenolt**
This restores behavior that was changed in c4a41ad which stopped `.0`
filling IP addresses correctly.

This change meant that entering in `10/8` as a subnet would end up being
parsed into `0.0.0.10/8` which is entirely different than `10.0.0.0/8`.

We are attempting to restore this behavior while still supporting
current behavior.

Fixes GH-184

### CODE REVIEW

File: sshuttle/options.py [hunk 1]
```diff
@@ -38,6 +40,15 @@ def parse_subnetport(s):
         raise Fatal('%r is not a valid address/mask:port format' % s)
 
     addr, width, fport, lport = m.groups()
+    # Attempt to 0 pad partial IPv4 address.
+    # 10 -> 10.0.0.0
+    if not ipv6:
+        try:
+            if addr.count('.') > 0 or int(addr) < 256:
```
Comment (mattrobenolt) [inline sshuttle/options.py]: This is a bit gross, but the only reasonable way I could figure out how to maintain current behavior to allow passing in `localhost` or an IP address as an integer, which is tested as `3098282570`.

**Comment by vieira** [comment]
Hello @mattrobenolt,

This breaks the ability to specify subnets as names, e.g. `sshuttle -r server example.com`.

It is also a bit inconsistent with other tools, like `ping`, `curl`, `ssh`, etc that treat `10.1` as `10.0.0.1` so it might be reasonable to expect that the `10.1` in `sshuttle -r server 10.1` is treated the same way as `10.1` in, e.g., `ping`.

With that said, I can see where you are coming from, however, I am not sure if the convenience of typing e.g. `10/8` instead of `10.0.0.0/8` is enough to justify to going back to doing the parsing ourselves.

AFAIK not even `iptables` allows that short hand syntax of using `10/8` to refer to `10.0.0.0/8`.

**Comment by mattrobenolt** [comment]
Yeah, I think you're completely right. The old behavior definitely doesn't match other tools, so I'm ok with keeping it as is. This change in behavior just threw me off for a while. It makes more sense how it is now.

Thanks for clarifying! ❤️

**Comment by brianmay** [comment]
Regardless, it is good that you have raised this as an issue. I will write a entry in the changelog for the next release to highlight this issue.

Thanks.

**Pull request closed without merging**